client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx

```diff
@@ -0,0 +1,286 @@
+import { Button } from '@wordpress/components';
+import { filterSortAndPaginate, Operator } from '@wordpress/dataviews';
+import { Icon, link, linkOff, plugins, trash } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { find } from 'lodash';
+import { useEffect, useMemo, useState } from 'react';
+import { initialDataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import ItemsDataViews from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews';
+import { navigate } from 'calypso/lib/navigate';
+import { useSelector } from 'calypso/state';
+import { Plugin } from 'calypso/state/plugins/installed/types';
+import { PluginActions } from '../hooks/types';
+import PluginActionStatus from '../plugin-management-v2/plugin-action-status';
+import { getPluginActionStatuses } from '../plugin-management-v2/utils/get-plugin-action-statuses';
+
+import './style.scss';
+
+interface Props {
+	currentPlugins: Array< Plugin >;
+	initialSearch?: string;
+	pluginsWithUpdates: Array< Plugin >;
+	activePlugins: Array< Plugin >;
+	inactivePlugins: Array< Plugin >;
+	isLoading: boolean;
+	onSearch?: ( search: string ) => void;
+	bulkActionDialog: ( action: string, plugins: Array< Plugin > ) => void;
+}
+
+export const PLUGINS_STATUS = {
+	ACTIVE: 1,
+	INACTIVE: 2,
+	UPDATE: 3,
+};
+
+export default function PluginsListDataViews( {
+	currentPlugins,
+	initialSearch,
+	pluginsWithUpdates,
+	activePlugins,
+	inactivePlugins,
+	isLoading,
+	onSearch,
+	bulkActionDialog,
+}: Props ) {
+	const allStatuses = useSelector( ( state ) => {
+		return getPluginActionStatuses( state );
+	} );
+
+	// Add flags for plugins status: active, inactive, updates
+	currentPlugins.map( ( plugin ) => {
+		plugin.status = [];
+
+		if ( find( pluginsWithUpdates, { slug: plugin.slug } ) ) {
+			plugin.status.push( PLUGINS_STATUS.UPDATE );
+		}
+
+		if ( find( inactivePlugins, { slug: plugin.slug } ) ) {
+			plugin.status.push( PLUGINS_STATUS.INACTIVE );
+		}
+
+		if ( find( activePlugins, { slug: plugin.slug } ) ) {
+			plugin.status.push( PLUGINS_STATUS.ACTIVE );
+		}
+
+		return plugin;
+	} );
```

Comment:
Can we optimize this? Feels like overkill to do it on every re-render. Should it live in state / selector territory?

In general mutating plugin objects like this feels wrong and potentially error prone.

Reply:
> Should it live in state / selector territory?

That makes sense! [Moved this to a selector here](https://github.com/Automattic/wp-calypso/pull/95561/commits/467f0d74fb956c73dbc54f73006ecc20ad579944).